File: ios/EmergencyApp/Sources/RegistrationViewModel.swift
import Foundation
import ParticipantRegistration

/// Guarda el perfil localmente y nada más — la subida real ocurre después,
/// disparada por `ConnectivityMonitor` dentro de `EmergencyViewModel`
/// (ADR-0003: desacoplada de este flujo, para que "continuar" nunca
/// requiera conectividad).
@MainActor
final class RegistrationViewModel: ObservableObject {
    @Published var name: String = ""
    @Published var contact: String = ""

    var canContinue: Bool {
        !trimmedName.isEmpty
    }

    private var trimmedName: String {
        name.trimmingCharacters(in: .whitespacesAndNewlines)
    }

    func completeRegistration() {
        let trimmedContact = contact.trimmingCharacters(in: .whitespacesAndNewlines)
        let profile = ParticipantProfile(name: trimmedName, contact: trimmedContact.isEmpty ? nil : trimmedContact)
        KeychainParticipantStore.save(profile)
    }
}

File: android/app/src/main/kotlin/com/farosos/app/RegistrationViewModel.kt
package com.farosos.app

import android.app.Application
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.setValue
import androidx.lifecycle.AndroidViewModel
import com.farosos.participantregistration.ParticipantProfile

/**
 * Guarda el perfil localmente y nada más — la subida real ocurre después,
 * disparada por `ConnectivityMonitor` dentro de `EmergencyViewModel`
 * (ADR-0003: desacoplada de este flujo, para que "continuar" nunca
 * requiera conectividad).
 */
class RegistrationViewModel(application: Application) : AndroidViewModel(application) {
    var name by mutableStateOf("")
    var contact by mutableStateOf("")

    val canContinue: Boolean
        get() = name.trim().isNotEmpty()

    fun completeRegistration() {
        val trimmedContact = contact.trim()
        val profile = ParticipantProfile(name = name.trim(), contact = trimmedContact.ifEmpty { null })
        ParticipantStore.save(profile, getApplication())
    }
}
